src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,104 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size;
 
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
     }
 
-    @Override
-    public void add(T value, int index) {
+    public ArrayList(int initialCapacity) {
+        values = (T[]) new Object[initialCapacity];
+    }
 
+    public void checkInBounds(int index) {
+        if (index > size - 1 || index < 0) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of bound!");
```

Comment:
here and in ohter invocations include index and bound in exception to make it more informative